src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -251,51 +285,95 @@ protected static boolean isRightOverhang(final GenomeLoc readLoc, final GenomeLo
      *
      * @param read                  the read bases
      * @param readStartIndex        where to start on the read
+     * @param readLength            the length of the read according to the reference, used to prevent overclipping
+     *                              softclipped output from SplitNCigarReads)
      * @param reference             the reference bases
      * @param referenceStartIndex   where to start on the reference
      * @param spanToTest            how many bases to test
      * @return true if too many overhanging bases mismatch, false otherwise
      */
     protected boolean overhangingBasesMismatch(final byte[] read,
                                                final int readStartIndex,
+                                               final int readLength,
                                                final byte[] reference,
                                                final int referenceStartIndex,
                                                final int spanToTest) {
         // don't process too small a span, too large a span, or a span that is most of a read
-        if ( spanToTest < 1 || spanToTest > MAX_BASES_IN_OVERHANG || spanToTest > read.length / 2 ) {
+        if ( spanToTest < 1 || spanToTest > maxBasesInOverhang || spanToTest > readLength / 2 ) {
             return false;
         }
 
         int numMismatchesSeen = 0;
         for ( int i = 0; i < spanToTest; i++ ) {
             if ( read[readStartIndex + i] != reference[referenceStartIndex + i] ) {
-                if ( ++numMismatchesSeen > MAX_MISMATCHES_IN_OVERHANG ) {
+                if ( ++numMismatchesSeen > maxMismatchesInOverhang) {
                     return true;
                 }
             }
         }
 
         // we can still mismatch overall if at least half of the bases mismatch
+
         return numMismatchesSeen >= ((spanToTest+1)/2);
     }
 
     /**
      * Close out the manager stream by clearing the read cache
      */
     public void close() {
```

Comment:
I think close should be renamed to `flush` since it doesn't actually close the stream.